net/netns/netns.go

```diff
@@ -66,3 +68,20 @@ type Dialer interface {
 	Dial(network, address string) (net.Conn, error)
 	DialContext(ctx context.Context, network, address string) (net.Conn, error)
 }
+
+func IsLocalhost(host string) bool {
+	name, _, err := net.SplitHostPort(host)
+	if err == nil {
+		host = name
+	} // error means the string didn't contain a port number, so ignore it.
+
+	if host == "localhost" || host == "localhost6" || host == "ip6-loopback" || host == "ip6-localhost" {
+		return true
+	}
+
+	ip, err := netaddr.ParseIP(host)
+	if err == nil && ip.IsLoopback() {
```

Comment:
could just `return err == nil && ip.IsLoopback()`

But the method call is well defined even on the ip zero value, so you could also just write:

```go
     ip, _ := netaddr.ParseIP(host)
     return ip.IsLoopback()
```

Reply:
Simplified.